### PR TITLE
Support both Marathon `1.4` and `1.5` API schemas

### PR DESCRIPTION
**Pull request by sschneid**
- `portMappings` under both `container.docker` and `container`
- new `networks` app-level array

### CODE REVIEW

File: src/test/groovy/mesosphere/marathon/client/model/v2/AppSpec.groovy [hunk 1]
```diff
@@ -150,162 +178,214 @@ class AppSpec extends Specification {
...
+		"name": "http",
+		"labels": {
+		  "vip": "192.168.0.1:80"
+		}
+	  }
+	]
+  },
+  "networks": [
+	{
+	  "mode": "container/bridge",
```
Comment (colin-msphere) [inline src/test/groovy/mesosphere/marathon/client/model/v2/AppSpec.groovy]: Should include a `name` attribute as well.

File: src/test/groovy/mesosphere/marathon/client/model/v2/AppSpec.groovy [hunk 2]
```diff
@@ -8,140 +8,168 @@ class AppSpec extends Specification {
...
+	def app = ModelUtils.GSON.fromJson(json, App.class)
+
+	expect:
+	// container.portMappings
+	app.container.portMappings[0].name == "http"
+	app.container.portMappings[0].protocol == "tcp"
+	app.container.portMappings[0].containerPort == 80
+	app.container.portMappings[0].servicePort == 10019
+
+	// networks
```
Comment (colin-msphere) [inline src/test/groovy/mesosphere/marathon/client/model/v2/AppSpec.groovy]: Should verify `name` as well.

Comment (sschneid) [reply src/test/groovy/mesosphere/marathon/client/model/v2/AppSpec.groovy]: Fixed.

File: src/test/groovy/mesosphere/marathon/client/model/v2/AppSpec.groovy (hunk 1 above)
Comment (sschneid) [reply src/test/groovy/mesosphere/marathon/client/model/v2/AppSpec.groovy]: Fixed.

**Pull request merged**